src/playsim/p_mobj.cpp

```diff
@@ -3351,52 +3352,106 @@ DEFINE_ACTION_FUNCTION(AActor, SetShade)
 	return 0;
 }
 
-void AActor::SetPitch(DAngle p, bool interpolate, bool forceclamp)
-{
-	if (player != NULL || forceclamp)
-	{ // clamp the pitch we set
-		DAngle min, max;
+// [MC] Helper function for Set(View)Pitch. 
+DAngle AActor::ClampPitch(DAngle p)
+{	
+	// clamp the pitch we set
+	DAngle min, max;
 
-		if (player != NULL)
-		{
-			min = player->MinPitch;
-			max = player->MaxPitch;
-		}
-		else
-		{
-			min = -89.;
-			max = 89.;
-		}
-		p = clamp(p, min, max);
+	if (player != nullptr)
+	{
+		min = player->MinPitch;
+		max = player->MaxPitch;
+	}
+	else
+	{
+		min = -89.;
+		max = 89.;
+	}
+	p = clamp(p, min, max);
+	return p;
+}
+
+void AActor::SetPitch(DAngle p, int fflags)
+{
+	if (player != nullptr || (fflags & SPF_FORCECLAMP))
+	{
+		p = ClampPitch(p);
 	}
+
 	if (p != Angles.Pitch)
 	{
 		Angles.Pitch = p;
-		if (player != NULL && interpolate)
+		if (player != nullptr && (fflags & SPF_INTERPOLATE))
 		{
 			player->cheats |= CF_INTERPVIEW;
 		}
 	}
+	
 }
 
-void AActor::SetAngle(DAngle ang, bool interpolate)
+void AActor::SetAngle(DAngle ang, int fflags)
 {
 	if (ang != Angles.Yaw)
 	{
 		Angles.Yaw = ang;
-		if (player != NULL && interpolate)
+		if (player != nullptr && (fflags & SPF_INTERPOLATE))
 		{
 			player->cheats |= CF_INTERPVIEW;
 		}
 	}
+	
 }
 
-void AActor::SetRoll(DAngle r, bool interpolate)
+void AActor::SetRoll(DAngle r, int fflags)
 {
 	if (r != Angles.Roll)
 	{
 		Angles.Roll = r;
-		if (player != NULL && interpolate)
+		if (player != nullptr && (fflags & SPF_INTERPOLATE))
+		{
+			player->cheats |= CF_INTERPVIEW;
+		}
+	}
+}
+
+void AActor::SetViewPitch(DAngle p, int fflags)
+{
+	if (player != NULL || (fflags & SPF_FORCECLAMP))
```

Comment:
@MajorCooke just a trivial styling issue but ``NULL`` should probably be changed to ``nullptr`` here